src/backend/joanie/core/models/products.py

```diff
@@ -90,6 +90,13 @@ class Product(parler_models.TranslatableModel, BaseModel):
         blank=True,
         null=True,
     )
+    contract_definition = models.ForeignKey(
+        "ContractDefinition",
+        verbose_name=_("contract definition"),
+        on_delete=models.PROTECT,
+        blank=True,
+        null=True,
+    )
```

Comment:
I agree that we may want to be able to do what you suggested... but for the moment we need to have a contract that is common to all organizations. So we may one day allow defining a custom contract definition per order group... but we still need a field on the product for the default cases? Otherwise you'll need to repeat it and it becomes counter productive? 

In particupar, for certificate products that are displayed on many courses, my feeling is that the contract is most likely a property of the product itself and doesn't vary for each course or each organization on which it is attached.
This does not prevent us from customising a contract based on a property of the organization. For example, add a field on the organization to specifcy if it is public or private and customize the contract template to take this into account on rendering?